apisix/plugins/authz-keycloak.lua

```diff
@@ -695,8 +703,107 @@ local function fetch_jwt_token(ctx)
     return token
 end
 
+-- To get new access token by calling get token api
+local function generate_token_using_password_grant(conf,ctx)
+    log.warn("generate_token_using_password_grant Function Called")
+    --Read Body
+    ngx.req.read_body()
+    --Get Body Data
+    local request_body=ngx.req.get_body_data()
+    local username = nil
+    local password = nil
+    --split by &
+    local parameters_array = util.split(request_body, "&")
```

Comment:
Why the username, password in the request body and use & to concat.

Reply:
As url encoded parameters were coming, they were automatically appended using '&'. Having said that, your comment helped me to utilize "ngx.decode_args" which makes code much cleaner. Thanks!